### ISSUE TITLE
Prometheus rule group provisioning - InternalServerError

### ISSUE THREAD

**Issue by UKFLW** [hashicorp/terraform-provider-azurerm#25066]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.5.5

### AzureRM Provider Version

3.65.0

### Affected Resource(s)/Data Source(s)

azurerm_monitor_alert_prometheus_rule_group

### Terraform Configuration Files

```hcl
## main.tf
resource "azurerm_monitor_alert_prometheus_rule_group" "promrulegroup" {
  count               = var.promrulegp_enabled == true ? 1 : 0
  name                = var.prg_name
  location            = var.prg_location
  resource_group_name = var.prg_rgname
  cluster_name        = var.prg_aksname
  description         = var.prg_description
  rule_group_enabled  = var.prg_rulegroup_enabled
  interval            = var.prg_interval
  scopes              = var.prg_scopes
  dynamic "rule" {
    for_each = var.prg_rule_enabled ? toset(var.prg_rule) : []
    content {
      alert       = rule.value["alert"]
      annotations = rule.value["annotations"]
      enabled     = rule.value["enabled"]
      expression  = rule.value["expression"]
      for         = rule.value["rule_for"]
      labels      = rule.value["labels"]
      severity    = rule.value["severity"]
      #record      = rule.value["record"]
      action {
        action_group_id   = rule.value["action_group_id"]
        action_properties = rule.value["action_properties"]
      }
      dynamic "alert_resolution" {
        for_each = rule.value.alert_res_enabled ? toset(rule.value.alert_res) : []
        content {
          auto_resolved   = alert_resolution.value["auto_resolved"]
          time_to_resolve = alert_resolution.value["time_to_resolve"]
        }
      }
    }
  }
  tags = var.prg_tags
}
```

### Debug Output/Panic Output

```shell
│ prometheusrulegroups.PrometheusRuleGroupsClient#CreateOrUpdate: Failure
│ responding to request: StatusCode=500 -- Original Error: autorest/azure:
│ Service returned an error. Status=500 Code="InternalServerError"
│ Message="There was a problem with the server. Please try again in a few
│ minutes. Activity ID: 55a2d0bb-e6b5-4440-b895-4ed3ec700bbf."
```

### Expected Behaviour

Prometheus Rule Group should be created with an alert.

### Actual Behaviour

module.prometheus-rule-group["prg1"].azurerm_monitor_alert_prometheus_rule_group.promrulegroup[0]: Still creating... [7m20s elapsed]
module.prometheus-rule-group["prg1"].azurerm_monitor_alert_prometheus_rule_group.promrulegroup[0]: Still creating... [7m30s elapsed]
╷
│ Error: creating Prometheus Rule Group (Subscription: "XXXXXXXXX"
│ Resource Group Name: "rg-XX-neu-platform-core-app-dev-01"
│ Prometheus Rule Group Name: "AKS_Pod_CPU_Percentage"): prometheusrulegroups.PrometheusRuleGroupsClient#CreateOrUpdate: Failure responding to request: StatusCode=500 -- Original Error: autorest/azure: Service returned an error. Status=500 Code="InternalServerError" Message="There was a problem with the server. Please try again in a few minutes. Activity ID: 55a2d0bb-e6b5-4440-b895-4ed3ec700bbf."
│ 
│   with module.prometheus-rule-group["prg1"].azurerm_monitor_alert_prometheus_rule_group.promrulegroup[0],
│   on .terraform/modules/prometheus-rule-group/prometheus-rule-group/main.tf line 1, in resource "azurerm_monitor_alert_prometheus_rule_group" "promrulegroup":
│    1: resource "azurerm_monitor_alert_prometheus_rule_group" "promrulegroup" {
│ 
│ creating Prometheus Rule Group (Subscription:
│ "XXXXXXXXX"
│ Resource Group Name: "rg-XX-neu-platform-core-app-dev-01"
│ Prometheus Rule Group Name: "AKS_Pod_CPU_Percentage"):
│ prometheusrulegroups.PrometheusRuleGroupsClient#CreateOrUpdate: Failure
│ responding to request: StatusCode=500 -- Original Error: autorest/azure:
│ Service returned an error. Status=500 Code="InternalServerError"
│ Message="There was a problem with the server. Please try again in a few
│ minutes. Activity ID: 55a2d0bb-e6b5-4440-b895-4ed3ec700bbf."

**Comment by ziyeqf** [comment]
Hi @UKFLW, thanks for reporting.

I have tried this resource on my machine but cannot reproduce the error. Could you please provide the log by setting `TF_LOG=DEBUG` environment variable? 

Thanks.

**Comment by UKFLW** [comment]
Thanks for your response and apologies for my delay in responding...

I have figured out the issue and able to resolve.
Issue was that value is not assigned to the "time_to_resolve" attribute (Under alert_resolution  block). Because in the TF Registry, it was specified as "Optional" - Refer https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/monitor_alert_prometheus_rule_group
Where as it is mandatory in the Azure Portal. Due to this conflict, the error we got "500 Internal Server Error" made further more confusion.

Finally, once the value is defined (ex: PT10M) to "time_to_resolve" variable, the issue got cleared and able to provision.
